### PR TITLE
Spring cloud stream functional bindings preview

### LINKED ISSUE #4196: Decouple from Spring Cloud Stream
The code should start using the new functional approach, replacing the deprecated annotations `@StreamListener` and `@EnableBindings`

Jira ref: AAE-6470

### PR DESCRIPTION
**Pull request by jsokolowskii**
This PR fixes Activiti/Activiti#4196 replacing the deprecated annotations from `spring-cloud-stream` with the new functional one provided by `spring-cloud-function`.

The annotation `@StreamListener` and `@EnableBindings` has been replaced with custom annotations defined in [Activiti-Cloud#910](https://github.com/Activiti/activiti-cloud/pull/910):
- `@FunctionalBinding`
- `@ConditionalFunctionalBinding`
- `@ConnectorBinding`

### CODE REVIEW

**Review by erdemedeiros — APPROVED** [review]
LGTM!

File: example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/ConnectorConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    @Override
+    public SubscribableChannel testErrorConnectorInput() {
+        return MessageChannels.publishSubscribe(TestErrorConnector.Channels.CHANNEL).get();
+    }
+
+    @Bean(INTERNAL_CHANNEL_PREFIX + RestConnector.Channels.POST)
+    @Override
+    public SubscribableChannel restConnectorPost() {
+        return MessageChannels.publishSubscribe(RestConnector.Channels.POST).get();
+    }
```
Comment (igdianov) [inline example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/ConnectorConfiguration.java:77]: Let's refactor this code using functional channel bindings.

File: example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/MultiInstanceConnector.java [hunk 2]
```diff
@@ -56,7 +55,12 @@ public MultiInstanceConnector(
         this.connectorProperties = connectorProperties;
     }
 
-    @StreamListener(value = Channels.CHANNEL)
+    @Override
+    public Void apply(IntegrationRequest event) {
```
Comment (igdianov) [inline example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/MultiInstanceConnector.java:59]: The `apply` method can be implemented directly.

File: example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/RestConnector.java [hunk 3]
```diff
@@ -48,7 +46,12 @@ public RestConnector(IntegrationResultSender integrationResultSender, ConnectorP
         this.connectorProperties = connectorProperties;
     }
 
-    @StreamListener(Channels.POST)
+    @Override
+    public Void apply(IntegrationRequest event) {
```
Comment (igdianov) [inline example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/RestConnector.java:50]: The `apply` method can be implemented directly.

File: example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/TestBpmnErrorConnector.java [hunk 4]
```diff
@@ -45,11 +45,15 @@ public TestBpmnErrorConnector(
     public interface Channels {
         String CHANNEL = "testBpmnErrorConnectorInput";
 
-        @Input(CHANNEL)
         SubscribableChannel testBpmnErrorConnectorInput();
     }
 
-    @StreamListener(value = Channels.CHANNEL)
+    @Override
+    public Void apply(IntegrationRequest event) {
```
Comment (igdianov) [inline example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/TestBpmnErrorConnector.java:52]: The `apply` method can be implemented directly.

File: example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/TestErrorConnector.java [hunk 5]
```diff
@@ -57,8 +55,13 @@ public TestErrorConnector(
         this.connectorProperties = connectorProperties;
     }
 
-    @StreamListener(value = Channels.CHANNEL)
-    public void handle(IntegrationRequest integrationRequest) throws InterruptedException {
+    @Override
+    public Void apply(IntegrationRequest event) {
```
Comment (igdianov) [inline example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/TestErrorConnector.java:59]: The `apply` method can be implemented directly.

File: example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/ExampleConnector.java [hunk 6]
```diff
@@ -49,34 +49,73 @@ public class ExampleConnector {
...
 
-    @StreamListener(value = ExampleConnectorChannels.EXAMPLE_CONNECTOR_CONSUMER)
-    public void performTask(IntegrationRequest event) throws InterruptedException {
+    @Override
+    public Void apply(IntegrationRequest event) {
+        performTask(event);
+        return null;
+    }
+
+    public void performTask(IntegrationRequest event) {
```
Comment (igdianov) [inline example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/ExampleConnector.java:75]: The apply method can be implemented directly or via separate Connector function bean that uses ExampleConnector consumer as a delegate.

File: example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/ExampleConnector.java [hunk 7]
```diff
@@ -25,21 +25,21 @@
...
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.cloud.stream.annotation.EnableBinding;
-import org.springframework.cloud.stream.annotation.StreamListener;
 import org.springframework.messaging.Message;
 import org.springframework.stereotype.Component;
 
-@Component
-@EnableBinding(ExampleConnectorChannels.class)
-public class ExampleConnector {
+@ConnectorBinding(input = ExampleConnectorChannels.EXAMPLE_CONNECTOR_CONSUMER, condition = "", outputHeader = "")
```
Comment (igdianov) [inline example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/ExampleConnector.java:40]: Why do you need to reset `outputHeader` attribute?

**Review by igdianov — CHANGES_REQUESTED** [review]

**Comment by jsokolowskii** [comment]
The changes were merged in PR #875

**Pull request closed without merging**